services/test_results.py

```diff
@@ -260,23 +243,44 @@ def build_message(self, payload: TestResultsNotificationPayload) -> str:
             results = f"Completed {completed} tests with **`{payload.failed} failed`**, {payload.passed} passed and {payload.skipped} skipped."
             message.append(results)
 
-        details = [
-            "<details><summary>View the full list of failed tests</summary>",
-            "",
-            "| **Test Description** | **Failure message** |",
-            "| :-- | :-- |",
-        ]
-
-        message += details
-
+        fail_dict = defaultdict(list)
+        flake_dict = defaultdict(list)
         for fail in payload.failures:
             flake = None
             if payload.flaky_tests is not None:
                 flake = payload.flaky_tests.get(fail.test_id, None)
-            test_description = self.generate_test_description(fail, flake)
-            failure_information = self.generate_failure_info(fail, flake)
-            single_test_row = f"| {test_description} | {failure_information} |"
-            message.append(single_test_row)
+
+            if flake is not None:
+                flake_dict[fail.testsuite].append(fail)
+            else:
+                fail_dict[fail.testsuite].append(fail)
+
+        def process_dict(d):
```

Comment:
Can this be made a helper function somewhere? It's hard to read functions within functions (even if it makes sense to be here and will never be reused)